$OpenBSD: patch-lib_awful_layout_suit_tile_lua_in,v 1.1 2014/11/11 12:20:16 dcoppa Exp $

commit 1f431384cbb428d97a6991ba0c59217cbc71b4a6
Author: Emmanuel Lepage Vallee <elv1313@gmail.com>
Date:   Mon Oct 13 15:07:31 2014 -0400

layouts: Allow layouts to be invoked with fake data

This patch allows 2 things to be done:

* Write unit test to validate layouts using fake clients and tags
* Query the current layout geometry from another tag

The advantages of the former are clear and simple. Those of the later include:

* Creating a screenshot of another layout
* Display the layout wireframe in the tag list (like KDE2-3, Gnome2)
* Having and 'ALT-tab' like visual popup for tags

--- lib/awful/layout/suit/tile.lua.in.orig	Fri Apr 11 11:07:10 2014
+++ lib/awful/layout/suit/tile.lua.in	Tue Nov 11 11:54:30 2014
@@ -74,7 +74,7 @@ local function tile_group(cls, wa, orientation, fact, 
 end
 
 local function do_tile(param, orientation)
-    local t = tag.selected(param.screen)
+    local t = param.tag or tag.selected(param.screen)
     orientation = orientation or "right"
 
     -- This handles all different orientations.
